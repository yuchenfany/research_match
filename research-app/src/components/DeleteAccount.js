/* eslint-disable */
/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */
import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import '../assets/index.css';

function DeleteAccount({ user, setUser }) {
  const [error, setError] = useState({ message: '' });
  const navigate = useNavigate();

  async function verify() {
    console.log(JSON.stringify(user));
    // console.log(JSON.stringify(getNextStudyID()));
    // setStudy({
    //   studyId: getNextStudyID(),
    // });
    const data = await fetch(`http://localhost:5000/record/delete/${user.username}`, {
      method: 'delete',
      headers: {
        'Content-Type': 'application/json',
      },
    });
    console.log('delete is being called');
    return true;
  }

  async function handleSubmit(event) {
    if (await verify()) {
      navigate('/');
    } else {
      event.preventDefault();
    }
}


  return (
    <div className="Create">
      <p className="header">Are you sure you want to delete your account?</p>

      <button
        type="button"
        className="goback-btn"
        onClick={async () => {
          navigate(`/${(user.type ?? 0) === 0 ? 'participant' : 'researcher'}-home`);
        }}
      >
        Go back
      </button>
      <div> </div>
      <button
        type="button"
        className="delete-btn"
        onClick={handleSubmit}
      >
        Delete account
      </button>
    </div>
  );
}

export default DeleteAccount;
