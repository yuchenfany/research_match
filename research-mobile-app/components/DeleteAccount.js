/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */

import React, { useState, useEffect } from 'react';
import { View, Button, Text, StyleSheet } from 'react-native';
// import NavBar from './NavBar';

function DeleteAccount({ route, navigation }) { // add props user
  let {user, setUser} = route.params;
  const[tempUser, setTempUser] = useState({
      username: user.username,
      password: user.password, // NOTE: unhashes password
      age: 0,
      // enrolled: user.enrolled,
      // age: user.age,
      // heightFeet: user.heightFeet,
      // heightInches: user.heightInches,
      // weight: user.weight,
      // sex: user.sex,
      // gender: user.gender,
      // allergies: user.allergies,
      // phys: user.phys,
      // psych: user.psych,
      // med: user.med,
      // type: user.type,
    }
  );

  async function verify() {
    console.log(JSON.stringify(user));
    // console.log(JSON.stringify(getNextStudyID()));
    // setStudy({
    //   studyId: getNextStudyID(),
    // });
    const data = await fetch(`http://localhost:5000/record/delete/${user.username}`, {
      method: 'delete',
      headers: {
        'Content-Type': 'application/json',
      },
    });
    console.log('delete is being called');
    return true;
  }

  async function handleSubmit(event) {
    if (await verify()) {
        navigation.navigate('Research Application', { user, setUser });
    } else {
      event.preventDefault();
    }
}

  
//   useEffect(() => {
//     getAllStudyJson()
//       .then(setEnrolledStudies);
//   }, []);

//   function goToStudy(studyId) {
//     console.log(studyId);
//     setStudy({ studyId });
//     navigate(`/study/${studyId}`);
//   }

  return (
    <View>
        <Text>Are you sure you want to delete your account ? </Text>
        <Button title="Delete" onPress={() => handleSubmit()}/>
        <Button title="Go back" onPress={() => navigation.push('ParticipantHome', { user, setUser })}/>
    </View>
  );
}

export default DeleteAccount;
