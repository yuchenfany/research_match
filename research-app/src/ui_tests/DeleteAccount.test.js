/**
 * @jest-environment jsdom
 */

/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import React from 'react';
// import { jest } from '@jest/globals';

import DeleteAccount from '../components/DeleteAccount';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

test('Are you sure? header', () => {
  render(
    <Router>
      <DeleteAccount />
    </Router>,
  );
  const header = screen.getByText('Are you sure you want to delete your account?');
  expect(header).toBeInTheDocument();
});

test('Go back button is clicked', () => {
  const mockSetUser = jest.fn();
  const mockHandleSubmit = jest.fn();
  const user1 = { username: 'user1', password: 'user1pass' };
  render(
    <Router>
      <DeleteAccount user={user1} setUser={mockSetUser} handleSubmit={mockHandleSubmit} />
    </Router>,
  );

  const button = screen.getByRole('button', {
    name: /go back/i,
  });

  fireEvent.click(button);
  expect(mockedUsedNavigate).toBeCalled();
});

test('Delete account button is clicked', () => {
  const mockSetUser = jest.fn();
  const mockHandleSubmit = jest.fn();
  const user1 = { username: 'user1', password: 'user1pass' };
  render(
    <Router>
      <DeleteAccount user={user1} setUser={mockSetUser} handleSubmit={mockHandleSubmit} />
    </Router>,
  );

  const button = screen.getByRole('button', {
    name: /delete account/i,
  });

  fireEvent.click(button);
  expect(mockSetUser).toBeCalled();
});

test('snapshot test', () => {
  const component = renderer.create(
    <Router>
      <DeleteAccount />
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
