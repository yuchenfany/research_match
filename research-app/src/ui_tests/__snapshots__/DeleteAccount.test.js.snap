// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="Create"
>
  <p
    className="header"
  >
    Are you sure you want to delete your account?
  </p>
  <button
    className="goback-btn"
    onClick={[Function]}
    type="button"
  >
    Go back
  </button>
  <button
    className="delete-btn"
    onClick={[Function]}
    type="button"
  >
    Delete account
  </button>
</div>
`;
